impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -492,6 +493,36 @@ public List<IGenericFile> getDeletedFiles() {
       .collect( Collectors.toList() );
   }
 
+  @Override
+  public void deleteFilePermanently( @NonNull GenericFilePath path ) throws OperationFailedException {
+    String fileId = getFileId( path );
+
+    try {
+      fileService.doDeleteFilesPermanent( fileId );
+    } catch ( Exception e ) {
+      throw new OperationFailedException( e );
+    }
+  }
+
+  protected String getFileId( @NonNull GenericFilePath path ) throws InvalidPathException {
```

Comment:
should be renamed to something like `getTrashFileId`, because the current name seems to indicate that we are getting an id from any kind of file